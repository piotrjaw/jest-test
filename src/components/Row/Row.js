import React from 'react';
import PropTypes from 'prop-types';

const Row = () => {

  return (
    <div />
  );
}

Row.PropTypes = {
};

export default Row;
