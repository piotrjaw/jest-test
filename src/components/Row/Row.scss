.jt-Row {
  &__text-checked {
    text-decoration: line -through;
  }
}
