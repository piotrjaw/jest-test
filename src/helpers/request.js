import axios from 'axios';
import omit from 'lodash.omit';

const headers = {
  'Accept': 'application/json',
  'Content-type': 'application/json'
};

const contructUrl = (url) => `http://localhost:5000${url}`;

export const del = (url, data) => axios.delete(contructUrl(url), { params: { ...data } });

export const get = (url, data) => axios.get(contructUrl(url), { params: { ...data } });

export const post = (url, data) => axios.post(contructUrl(url), data);

export const put = (url, data) => {
  const { _id } = data;
  const requestData = omit(data, '_id');
  return axios.put(contructUrl(url), requestData, {
    params: { _id },
    headers
  });
};
