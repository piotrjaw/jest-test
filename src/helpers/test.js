import React from 'react';

export const getWrapper = ({ Component, method, props }) => {
  return method(
    <Component
      { ...props }
    >
      { props.children }
    </Component>
  );
};

export const getMockStore = (mockFn, state = {}) => ({
  subscribe: mockFn(),
  dispatch: mockFn(),
  getState: mockFn(() => ({
    ...state,
  })),
});
