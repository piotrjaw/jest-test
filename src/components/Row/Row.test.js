import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { getWrapper } from '../../helpers/test';
import Row from './Row';

const setupWrapper = (config = {}) => {
  const { method = shallow, props = {} } = config;
  return getWrapper({
    Component: Row,
    method,
    props: {
      ...props
    }
  });
};

let wrapper;

beforeEach(() => {
  wrapper = setupWrapper();
  jest.resetAllMocks();
});

afterEach(() => {
  expect(toJSON(wrapper.render())).toMatchSnapshot();
  wrapper = null;
});

describe('Row', () => {
  it('should render', () => {
  });

  it('should render a to-do text', () => {
  });

  it('should render a button', () => {
  });

  it('should handle a checked entry', () => {
  });

  it('should handle a unchecked entry', () => {
  });

  it('should render a `Delete` button', () => {
  });

  it('should call onCheckClick with _id and negation of current checked status when clicked', () => {
  });

  it('should call onDeleteClick with _id when clicked', () => {
  });
});
