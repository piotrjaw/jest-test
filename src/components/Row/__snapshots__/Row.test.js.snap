// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should call onCheckClick with _id and negation of current checked status when clicked 1`] = `<div />`;

exports[`Row should call onDeleteClick with _id when clicked 1`] = `<div />`;

exports[`Row should handle a checked entry 1`] = `<div />`;

exports[`Row should handle a unchecked entry 1`] = `<div />`;

exports[`Row should render 1`] = `<div />`;

exports[`Row should render a \`Delete\` button 1`] = `<div />`;

exports[`Row should render a button 1`] = `<div />`;

exports[`Row should render a to-do text 1`] = `<div />`;
